tests/runtime/in_syslog.c

```diff
@@ -305,6 +305,61 @@ static int init_udp(char *in_host, int in_port, struct sockaddr_in *addr)
     return fd;
 }
 
+/* Copy src into dst, stripping a single trailing '\n' if present; return len */
+static size_t rstrip_nl_copy(char *dst, size_t dstsz, const char *src)
+{
+    size_t n = strlen(src);
+    if (n > 0 && src[n - 1] == '\n') {
+        n -= 1;
+    }
+    if (n + 1 > dstsz) {
+        n = dstsz - 1;
+    }
+    memcpy(dst, src, n);
+    dst[n] = '\0';
+    return n;
+}
+
+/* Build one octet-counted frame into 'out' as: "<len> " + msg [+ '\n' if add_lf] */
+/* Returns total bytes written (excluding terminal '\0' in 'out') */
+static size_t build_octet_frame(char *out, size_t outsz,
+                                const char *msg, int add_lf)
+{
+    char tmp[2048];
+    size_t mlen = rstrip_nl_copy(tmp, sizeof(tmp), msg);
```

Comment:
do not do this type of assignations move them after the declarations so get easy to read

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/11035/commits/e3c8ba83b44a399f7aec311e0a014964a9fab07c.